paper.tex

```diff
@@ -64,6 +64,20 @@ \section{Related Work}
 
 
 \section{System Architecture}
+\subsection{Participants}
+There are four major roles in the decentralized data marketplace.
+
+\subsubsection{Registrar}
+Registrar is responsible for creating the Registration Contract and adding new data providers, consumers and brokers to the decentralized data marketplace.
+
+\subsubsection{Provider}
+Data providers are sensor owners who are willing to sell sensing data to consumers. They decide the subscription period, subscription price, data type, sampling frequency. Afterward, they can launch the product on the decentralized data marketplace. Next, they collect the raw data with their sensors and send it to the broker who will assist in uploading it to the network. A MAM channel\cite{MAM} is used to store the encrypted data stream.
```

Comment:
Replace "sensor owners" with appropriate subjects which generate and preserve data.